src/ore/src/vec.rs

```diff
@@ -77,6 +77,121 @@ where
     }
 }
 
+/// Extention methods for `std::vec::Vec`
+pub trait VecExt<T> {
+    /// Creates an iterator which uses a closure to determine if an element should be removed.
+    ///
+    /// If the closure returns true, then the element is removed and yielded.
+    /// If the closure returns false, the element will remain in the vector and will not be yielded
+    /// by the iterator.
+    ///
+    /// Using this method is equivalent to the following code:
```

Comment:
This is subjective, but you may want to include the call to `drain_filter_swapping` that would be equivalent.

Reply:
Ah, good point. I borrowed the docs from `drain_filter` in std https://doc.rust-lang.org/stable/std/vec/struct.Vec.html#method.drain_filter

Since CI is already green I'll merge as-is if that's ok